### DISCUSSION TITLE
[Question]Why is the default value of max_prefill_tokens 16384?

### DISCUSSION THREAD

**Discussion by wjj19950828** [sgl-project/sglang#1468] [Q&A] [answered]
Why is the default value of max_prefill_tokens 16384? How is it set?

Is this parameter similar to ```max_num_batched_tokens``` in vLLM? For example, if it is a 32k model, does it need to be adjusted to 32k? If I adjust it to 32k, will there be an OOM problem?

Thanks~

**Comment by wjj19950828** [comment]
@zhyncs Do you have any suggestions? Thanks~

**Comment by merrymercy** [comment, marked as answer]
You do not need to worry about this.
By default, sglang uses chunked prefill with a chunk size of 8k https://github.com/sgl-project/sglang/blob/13f1357ef000e0d9dcf6f13dd178d809126f3ac7/python/sglang/srt/server_args.py#L323-L328.

If you have a 32k model, it will run it chunk by chunk, so no any addition settings are required to support it.

`max_prefill_tokens` is a legacy config for the cases where chunked prefill is not enabled. In that case, you need to update it to 32k for your 32k model
